adagrams/game.py

```diff
@@ -1,11 +1,113 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_CHART = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
+
 def draw_letters():
-    pass
+    new_pool = []
+    for letter, count in LETTER_POOL.items():
+        new_pool.extend([letter] * count)
+    
+    letters = []
+    while len(letters) < 10:
+        draw = random.randrange(0, len(new_pool))
+        letter = new_pool.pop(draw)
+        letters.append(letter)
+
+    return letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    uppercase_word = word.upper()
+    bank_copy = list(letter_bank)
+    
+    for char in uppercase_word:
+        if char in bank_copy:
```

Comment:
Since bank_copy is a list of letters, the time complexity of checking if a `char` is in the bank_copy is O(n). You need to loop through every letter in bank_copy to find the letter. You can optimize this slightly by creating a letter_frequency dictionary instead of bank_copy where the keys are the letters and the values are frequency of the letter. 

Then when you check if a char is a key of letter_frequency, you reduce your time complexity to O(1). 
```python
if char in letter_frequency:
```

Take care to use `key in dict` syntax. If you do `key in dict.keys()` you'd reproduce O(n) time complexity since under the hood Python creates a list of keys when you call .keys() and the `in` operator for lists is linear.